Add pen support for PolygonBrush

rvankevelaer opened the pull request:
- Added pressure-sensitive pen support for PolygonBrush and ModifyPolygonBrush
- The sketch radius is scaled between the minimum brush size and the current absolute brush size
- The absolute size of the sketch feature is shown
- A penOnly condition is added in events/condition.js
- The PolygonBrush example is cleaned up

Commit:
Add pen freehand mode for PolygonBrush

Commit:
Add pen support for PolygonBrush

- Add pen support for PolygonBrush and ModifyPolygonBrush
- Add condition penOnly to events/conditions.js

Commit:
Remove unnecessary imports from Polygon Brush example

mzur commented on src/ol/interaction/ModifyPolygonBrush.js:
```diff
@@ -103,6 +110,25 @@ class ModifyPolygonBrush extends Modify {
+  updateSketchRadiusByPressure_(event) {
+    if (this.sketchPoint_ && event.pointerEvent.pressure != 0) {
+      this.sketchPointRadius_ = getNewSketchPointRadius(event, this.sketchPointRadius_);
+      this.sketchRadius_ = Math.max(
```
You could add a new `getNewSketchPointRadiusByPressure` function to the `PolygonBrush` file and import it here just as `getNewSketchPointRadius`. This way you don't have to duplicate code (e.g. this function and `MIN_BRUSH_SIZE`).

mzur commented on src/ol/interaction/ModifyPolygonBrush.js:
```diff
@@ -103,6 +110,25 @@ class ModifyPolygonBrush extends Modify {
+  getSketchGeometry(event) {
+    var sketchGeometry  = this.sketchPoint_.getGeometry();
+    if (event.originalEvent.pointerType === 'pen') {
+      sketchGeometry = this.sketchPoint_.clone().getGeometry();
```
This probably has a negative impact on performance (we spoke about this). Instead, create an invisible second sketchPoint that has the radius which was determined by pen pressure and update only the radius.

mzur commented on src/ol/interaction/ModifyPolygonBrush.js:
```diff
@@ -103,6 +110,25 @@ class ModifyPolygonBrush extends Modify {
+      sketchGeometry = this.sketchPoint_.clone().getGeometry();
+      sketchGeometry.setRadius(this.sketchRadius_);
+    }
+    return sketchGeometry;
```
Please add an empty line above `return` statements.

mzur commented on src/ol/interaction/PolygonBrush.js:
```diff
@@ -165,12 +176,32 @@ class PolygonBrush extends Draw {
+  getSketchGeometry(event) {
+    var sketchGeometry  = this.sketchPoint_.getGeometry();
+    if (event.originalEvent.pointerType === 'pen') {
+      sketchGeometry = this.sketchPoint_.clone().getGeometry();
```
see above

mzur commented on src/ol/interaction/PolygonBrush.js:
```diff
@@ -165,12 +176,32 @@ class PolygonBrush extends Draw {
+  updateSketchRadiusByPressure_(event) {
+    if (this.sketchPoint_ && event.pointerEvent.pressure != 0) {
+      this.sketchPointRadius_ = getNewSketchPointRadius(event, this.sketchPointRadius_);
+      this.sketchRadius_ = Math.max(
```
see above

mzur commented on src/ol/interaction/PolygonBrush.js:
```diff
@@ -210,6 +242,17 @@ class PolygonBrush extends Draw {
     return this.sketchPointRadius_;
   }
+
+  abortDrawing_() {
```
Don't override this function completely. Instead, call `super()` and only delete the new invisible sketchPoint here (if `sketchFeature` is `true`.

mzur reviewed (changes requested):
We spoke about some of my comments already. I've added them so I don't forget about anything.

Commit:
Fix speed issues

- Remove feature clones
- sketchCircle as class variable
- Add buildung biigle/openlayers in DEVELOPING.md

mzur reviewed (approved):
I have applied some [minor changes](https://github.com/biigle/openlayers/commit/efdd5dc0185611f316c83ae3c1386cc09207b1a1) but couldn't push them to your fork. Please test this version (after the merge) again. Then we can add the new feature to BIIGLE.
